pyhms/cluster/kriging.py

```diff
@@ -0,0 +1,55 @@
+import numpy as np
+import plotly.graph_objects as go
+from pykrige.ok import OrdinaryKriging
+
+from ..core.population import Population
+
+
+class KrigingLandscapeApproximator:
+    def __init__(self) -> None:
+        self.population: Population | None = None
+        self.model: OrdinaryKriging | None = None
+
+    def fit(self, population: Population) -> None:
+        self.population = population
+        if population.genomes.shape[1] != 2:
+            raise ValueError("Kriging only supports 2-dimensional genomes")
+        self.model = OrdinaryKriging(
+            population.genomes[:, 0],
+            population.genomes[:, 1],
+            population.fitnesses,
+            variogram_model="linear",
+            verbose=False,
+            enable_plotting=False,
+        )
+
+    def predict(self, genomes: np.ndarray) -> np.ndarray:
+        return self.model.execute("points", genomes)
+
+    def plot(self, number_of_points_per_dim: int = 100) -> None:
```

Comment:
This works only in two dimensions, so there should be some assertion on bounds shape. The clear message will make it less confusing for the user.

Reply:
to plot you need to fit, there's if:
```
if population.genomes.shape[1] != 2:
            raise ValueError("Kriging only supports 2-dimensional genomes")
```
but sure, I can add the message here as well

Reply:
Oh, I missed that. Nevermind then